src/App.js

```diff
@@ -14,11 +15,91 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  }
+
+  getTitle = () => {
+    const { selectedGoods } = this.state;
+
+    if (selectedGoods.length === 1) {
+      return `${selectedGoods[0]} is selected`;
+    }
+
+    if (selectedGoods.length > 1) {
+      return `${selectedGoods.join(', ')} are selected`;
+    }
+
+    return 'No good selected';
+  }
+
+  addProducts = (product) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, product],
+    }));
+  };
+
+  removeProduct = (product) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== product),
+    }));
+  };
+
+  removeProductAll = () => {
+    this.setState({ selectedGoods: [] });
+  }
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <div className="header">
+          <h1 className="Product__title">
+            {this.getTitle()}
+          </h1>
+          <button
+            type="button"
+            onClick={this.removeProductAll}
+            className="Product__title-button"
+          >
+            X
+          </button>
+        </div>
+        <ul className="Product__list">
+          {goodsFromServer.map(product => (
+            <li
+              key={product}
+              className={classNames('Product__item', {
+                Product__item_added: selectedGoods.includes(product),
```

Comment:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once